QUESTIONS.md

```diff
@@ -0,0 +1,297 @@
+        const number = 10;
+    }
+}
+
+// Number cannot be used here
+```
+
+**When should I use *for each* vs *map*?**
+
+Map is used to transform each element of an array or either an object, it returns a new array 
+
+*Example using 'map':*
+
+```
+const array = [1, 2, 3, 4]
+
+array.map(function(e) {
+    return e * 2
+})
+
+console.log(array)
+
+El resultado será:
+[2, 4, 6, 8]
+
+```
+
+forEach is used to loop each element of an array or object 
+
+*Example using 'forEach':*
+
+```
+array.forEach(function(e) {
+    console.log(e * 2)
+})
+
+El resultado será:
+2
+4
+6
+8
+```
+
+**What is the *reduce* use for?**
+
+Is used to make calculations and it returns a value. The main idea of use reduce is to have a map and filter working together, you can filter the data that you have and map it in the same step
+
+*Example using 'reduce':*
+
+```
+const array = [1, 2, 3, 4]
+
+const initialValue = 10;
+
+const sumWithInitial = array.reduce((aux, current); => aux + current, initialValue
+);
+
+console.log(sumWithInitial)
+```
+
+**What is the difference between regular functions and arrow functions?**
+
+There are many difference that we can find between these type of functions, here are some difference: 
+
+|                         | Regular functions                                                                                                         | Arrow functions                                                                                                                                                                                        |
+|-------------------------|---------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Difference way to write | ``` function sum(a, b) {    return a + b  } ``` <br> <br> ``` const sum = function sum(a, b) {    return a + b  } ```     | ``` const sum = (a, b) => {   return a + b } ``` <br> <br> ```  const sum = (a, b) => a + b ```                                                                                                        |
+| Arguments object        | ``` function regularFun() {    return Math.max(...arg); } ```                                                             | ``` const arrowFun = () => {   // If we try to access the arg    we're gonna get an exception } ``` <br> <br> // Right way to do <br> <br> ``` const arrowFun = (...numbers) =>  Math.max(numbers) ``` |
+| Prototype object        | ``` function regularFun() {} ``` <br> <br> ``` console.log(regularFun.prototype) ``` <br> <br> ``` { constructor: f } ``` | ``` const arrowFun = () => {} ``` <br> <br> ``` console.log(arrowFun.prototype)  ``` <br> <br> ``` undefined ```                                                                                       |
+
```

Comment:
Good examples, just to clarify, the main difference is that regular functions has a context `this`, but arrow functions does not have a context. Because of that, arrow function does not have a `Prototype`, and your last example returns undefined 👍 .